components/badges/RequirementRow.jsx

```diff
@@ -0,0 +1,168 @@
+var React = require('react');
+
+var RequirementRow = React.createClass({
+  propTypes: {
+    position: React.PropTypes.number.isRequired,
+    text: React.PropTypes.string.isRequired,
+    icon: React.PropTypes.oneOfType([
+      React.PropTypes.bool,
+      React.PropTypes.string
+    ])
+  },
+
+  getInitialState: function() {
+    return {
+      evidenceFiles: [],
+      renderEvidenceFields: false
+    };
+  },
+
+  render: function () {
+    var icon = null;
+
+    if (this.props.icon !== false) {
+      icon = this.props.icon ? this.props.icon : 'fa fa-check';
+    }
+
+    return (
+      <li>
+        <div className="icon-wrapper">
+          <span className={icon}></span>
+        </div>
+        <div className={"requirement" + (this.state.renderEvidenceFields ? " expanded" : "")}>
+          <div className="task-name">{ this.props.text }</div>
+          <div className="evidence">{ this.renderEvidenceOptions()}</div>
+        </div>
+      </li>
+    );
+  },
+
+  renderEvidenceOptions: function() {
+    if (!this.props.evidence) {
+      return null;
+    }
+
+    if (!this.state.renderEvidenceFields) {
+      return (
+        <button className="btn" onClick={() => this.setState({ renderEvidenceFields: true })}>Submit evidence for this task</button>
+      );
+    }
+
+    return (
+      <div className="apply-send-qualifications">
+        {/*
+          <div className="evidence-title">{ this.props.evidence }</div>
+        */}
+        <div className="horizontal-form">
+          <fieldset>
+            <label className="control-label">Document your evidence here:</label>
+            <textarea
+              rows={10}
+              onChange={this.updateEvidenceText}
+              value={this.state.evidenceText}
+            />
+          </fieldset>
+
+          <fieldset>
+            <input type="file" className="hidden" ref="optionalFile" onChange={this.handleFiles}/>
+            <label className="control-label">Attach one or more file (if needed):</label>
+            <button className="btn attach" onClick={this.selectFiles}>click here to pick one or more files...</button>
+            { this.generateAttachmentSelection() }
+          </fieldset>
+        </div>
+      </div>
+    );
+  },
+
+  generateAttachmentSelection: function() {
+    if (this.state.evidenceFiles.length === 0) {
+      return null;
+    }
+
+    return this.state.evidenceFiles.map(e => {
```

Comment:
no, because evidence is the full collection of fields. Both the text and the files are evidence as far as Credly cares.